kolena/_experimental/object_detection/evaluator_single_class.py

```diff
@@ -163,11 +167,8 @@ def compute_test_case_metrics(
         recall = compute_recall(tp_count, fn_count)
         f1_score = compute_f1_score(tp_count, fp_count, fn_count)
 
-        average_precision = 0.0
-        if precision > 0:
-            baseline_pr_curve = compute_pr_curve(all_bbox_matches)
-            if baseline_pr_curve is not None:
-                average_precision = compute_average_precision(baseline_pr_curve.y, baseline_pr_curve.x)
+        if average_precision is None:
+            average_precision = compute_average_precision([precision], [recall])
```

Comment:
I'm not sure if I follow this change -- why are we attempting to compute AP with a single `([precision], [recall])` point? Is this a previously missed edge case?

Reply:
It's also a bit of smell to be attempting to compute it in two steps that are in different functional steps (`test_case_metrics_single_class` and `compute_test_case_metrics`) -- makes the computation very hard to follow

Reply:
>I'm not sure if I follow this change -- why are we attempting to compute AP with a single ([precision], [recall]) point? Is this a previously missed edge case?

Removed this case.  With the recent change in `_compute_threshold_curves` handling a single datapoint case, `None` `average_precision` case check is unnecessary. Adjusted some unit tests to accommodate this change.

Reply:
>It's also a bit of smell to be attempting to compute it in two steps that are in different functional steps (test_case_metrics_single_class and compute_test_case_metrics) -- makes the computation very hard to follow

@marko-polo-cheno can take a look at this comment once he is back.

Reply:
Your latest change also gets rid of the second computation, which sorts out the code-smell 👍

Reply:
I computed AP in 2 steps:
  1. from the PR curve, get the AP from the curve if it exists
  2. if the PR curve doesn't exist (e.g. only one point), we calculate AP with that single `([precision], [recall])` point
This complication only exists to handle the one-point PR curve case, which is hardly ever seen.